app/upgrader.go

```diff
@@ -22,13 +23,13 @@ func upgradeLonghornRelatedComponents(kubeClient *clientset.Clientset, namespace
 	if err != nil {
 		return err
 	}
-	for _, pv := range pvList.Items {
+	for i, pv := range pvList.Items {
 		// Make sure we are handling the Longhorn related PVs only
 		if v, exist := pv.Annotations[PVAnnotationCSIProvisioner]; !exist || v != types.DeprecatedProvisionerName {
 			continue
 		}
 		pv.Annotations[PVAnnotationCSIProvisioner] = types.LonghornDriverName
```

Comment:
It seems that we need to modify this as well:
```
pvList.Items[i].Annotations[PVAnnotationCSIProvisioner] = types.LonghornDriverName
```

or using this kind of format:
```
for i := range pvList.Items {
	pv := &pvList.Items[i]
	if v, exist := pv.Annotations[PVAnnotationCSIProvisioner]; !exist || v != types.DeprecatedProvisionerName {
		continue
	}
	pv.Annotations[PVAnnotationCSIProvisioner] = types.LonghornDriverName

	if _, err := kubeClient.CoreV1().PersistentVolumes().Update(context.TODO(), pv, metav1.UpdateOptions{}); err != nil {
		return err
	}
```

The key point of these modifications is avoiding using the pointer of the temporary variable initialized by the "for" loop.